### ISSUE TITLE
Could it be possible to create a new DI scope for handlers ?

### ISSUE THREAD

**Issue by wuuer** [martinothamar/Mediator#88]
The original MediatR does it

**Comment by Tornhoof** [comment]
Where does the original MediatR do that?

**Comment by martinothamar** [comment]
Yes I don't think MediatR does this. atleast I can't find a reference to `CreateScope`/`CreateAsyncScope`, and can't remember there ever being any from the previous times I've went over the MediatR code. For most applications I imagine ASP.NET Core handles creation and disposing of scopes, if not it can always be done manually

**Comment by wuuer** [comment]
> Yes I don't think MediatR does this. atleast I can't find a reference to `CreateScope`/`CreateAsyncScope`, and can't remember there ever being any from the previous times I've went over the MediatR code. For most applications I imagine ASP.NET Core handles creation and disposing of scopes, if not it can always be done manually

> Where does the original MediatR do that?

in the Publishing strategies code:  https://github.com/jbogard/MediatR/discussions/736#discussioncomment-5422614
btw ,when will this library support customing the Publishing strategies?

**Comment by Tornhoof** [comment]
So MediatR doesn't do it, but some extended code in the blog post in that discussion.

**Comment by wuuer** [comment]
> So MediatR doesn't do it, but some extended code in the blog post in that discussion.

Right.
So here is the question:
How to customize the Publishing strategies like MediatR.

**Comment by martinothamar** [comment]
Publishing strategies haven't been implemented here. It's going to be pretty far down on my list for now unfortunately, unless someone else wants it enough to make a PR

**Comment by AeonLucid** [comment]
Just posting my 2 cents, I had the same problem. Originally I thought my dependencies would be scoped to the handler but after a long time I just discovered I was wrong after chasing a bug. I was injecting `IMediator` into my Blazor Server pages and components and the handlers were all sharing the same scoped dependencies. Which is bad in blazor because scoped means scoped to the websocket connection there.

I managed to create a solution which is similar to these two comments on MediatR.
- https://github.com/jbogard/MediatR/issues/607#issuecomment-1208027814
- https://github.com/jbogard/MediatR/issues/607#issuecomment-1312616013
